### PR TITLE
Add a simple script to setup a local Tekton cluster

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Add a script that uses kind to setup a local Tekton cluster with
the latest releases of pipeline, triggers and dashboard or
specific versions if specified.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

/kind feature

### CODE REVIEW

**Comment by sbwsg** [comment]
Is this something we plan to give support / handle issues for?  If not, I wonder if adding it in [a `contrib` directory](https://softwareengineering.stackexchange.com/questions/252053/whats-in-the-contrib-folder) might be a better signal of that?  I've always thought of `hack` as our "knative scripts" directory.  Although I just read the README in `hack` and it does say that it's a place for development scripts.  So I'm on the fence.

Script itself lgtm!

**Comment by bobcatfish** [comment]
> I've always thought of hack as our "knative scripts" directory. 

The hack folder has always made me sad; the name implies that the stuff inside is just hacks, which seems to imply that we won't keep it up to good standards and also seems like a naming short cut ala "lib" or "helpers".

My 2 cents not but not anything actionable for this PR XD

**Comment by vdemeester** [comment]
> The hack folder has always made me sad; the name implies that the stuff inside is just hacks, which seems to imply that we won't keep it up to good standards and also seems like a naming short cut ala "lib" or "helpers".

Ah interersting, I tend to prefer `hack` than `scripts` :stuck_out_tongue: and I didn't really though of it as "just hacks, no good standards, …" :sweat_smile:.

**Comment by afrittoli** [comment]
> > I've always thought of hack as our "knative scripts" directory.
> 
> The hack folder has always made me sad; the name implies that the stuff inside is just hacks, which seems to imply that we won't keep it up to good standards and also seems like a naming short cut ala "lib" or "helpers".
> 
> My 2 cents not but not anything actionable for this PR XD

Heh, I used the `hack` folder thinking that includes files that are used for development (hacking)... but I don't mind really where it leaves.  I added docs to `DEVELOPMENT.md` which hopefully will make it clear what the script is about and how to use it.

**Comment by afrittoli** [comment]
The instructions worked well on my laptop - but I'm on a mac. Would anyone running Linux mind trying out this setup? For me it allowed me to get a local dev env with CI running pretty quickly!

**Comment by afrittoli** [comment]
> Is this something we plan to give support / handle issues for? If not, I wonder if adding it in [a `contrib` directory](https://softwareengineering.stackexchange.com/questions/252053/whats-in-the-contrib-folder) might be a better signal of that? I've always thought of `hack` as our "knative scripts" directory.

The main goal I'd like to achieve with this is to make it easy for new contributors to get a local Tekton dev cluster running - and hopefully foster contributions :)
I don't mind what the folder is,  as long as it's documented in development.md

**Comment by sbwsg** [comment]
It doesn't sound like we have consensus around that idea and I don't feel strongly.  This can always be changed in future too.

/lgtm

File: hack/tekton_in_kind.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# Setup the kind cluster
+kind create cluster --name tekton
+
+# Install Tekton
+kubectl apply -f https://storage.googleapis.com/tekton-releases/pipeline/previous/${TEKTON_PIPELINE_VERSION}/release.yaml
+kubectl apply -f https://storage.googleapis.com/tekton-releases/triggers/previous/${TEKTON_TRIGGERS_VERSION}/release.yaml 
+kubectl apply -f https://github.com/tektoncd/dashboard/releases/download/${TEKTON_DASHBOARD_VERSION}/tekton-dashboard-release.yaml
+
+# Wait just a bit, to wait for the service to be running
+sleep 10
```
Comment (afrittoli) [inline hack/tekton_in_kind.sh:69]: This needs to be fixed with a proper wait...

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -1,3 +1,97 @@
...
+GITHUB_TOKEN=<set-the-token-here>
+GITHUB_SECRET=$(ruby -rsecurerandom -e 'puts SecureRandom.hex(20)')
+GITHUB_ORG=<org> # The org or user where your fork is hosted
+GITHUB_REPO=<repo> # The name of the fork, typically "plumbing"
+GITHUB_USER=<github-user> # Your GitHub username
+
+# Deploy plumbing resources. Run from the root of your local clone
+# The sed command injects your fork GitHub org in the CEL filters
+kustomize build tekton/ci | \
+  sed -E 's/tektoncd(\/p[^i]+|\(|\/'\'')/'$GITHUB_ORG'\//g' | \
```
Comment (afrittoli) [inline DEVELOPMENT.md:64]: Almost...
```
 sed -E 's/tektoncd(\/p[^i]+|\(|\/'\'')/'$GITHUB_ORG'\1/g'
```

**Review by vdemeester — APPROVED** [review]
/meow

File: hack/tekton_in_kind.sh (hunk 1 above)
Comment (AlanGreene) [reply hack/tekton_in_kind.sh:69]: Something like this?
`kubectl wait -n tekton-pipelines --for=condition=ready pods --all`

We use a variation on this in the dashboard walkthrough with kind

**Pull request merged**